org.eclipse.search/search/org/eclipse/search/internal/core/text/TextSearchVisitor.java

```diff
@@ -528,9 +528,18 @@ private List<TextSearchMatchAccess> locateMatches(IFile file, CharSequence searc
 				ReusableMatchAccess access= new ReusableMatchAccess();
 				access.initialize(file, start, end - start, searchInput);
 				occurences.add(access);
-				boolean res= fCollector.acceptPatternMatch(access);
-				if (!res) {
-					return occurences; // no further reporting requested
+				try {
+					boolean res = fCollector.acceptPatternMatch(access);
+					if (!res) {
+						return occurences; // no further reporting requested
+					}
+				} catch (CoreException e) {
+					if (e.getStatus() == Status.CANCEL_STATUS) {
```

Comment:
Better is to catch `OperationCanceledException` which is what probably should be thrown